src/msemu.c

```diff
@@ -897,7 +897,7 @@ int main(int argc, char *argv[])
 			} else if (execute_counter > 15) {
 				execute_counter = 0;
 				while (tstate_counter < interrupt_period ||
-				  !z80ex_int_possible(ms.z80)) {
+				  z80ex_last_op_type(ms.z80)) {
```

Comment:
Add comments to explain this while loop. I think they already existed but probably forgot to change them when this was fixed.